packages/@glimmer/component/src/-private/component.ts

```diff
@@ -0,0 +1,286 @@
+import { DEBUG } from '@glimmer/env';
+
+const DESTROYING = new WeakMap<GlimmerComponent<object>, boolean>();
```

Comment:
I know this is copied from the the old repo -- but I feel like we could use the destroyable primitives instead (which i'm more than happy to PR after this (#20751) lands)

```js
export {
  assertDestroyablesDestroyed,
  associateDestroyableChild,
  destroy,
  enableDestroyableTracking,
  isDestroying,
  isDestroyed,
} from '@glimmer/destroyable';

import { registerDestructor } from '@glimmer/destroyable';
```

https://github.com/glimmerjs/glimmer-vm/blob/main/packages/%40glimmer/destroyable/index.ts#L204

Reply:
Yeah that makes sense as a followup refactor.